### PR TITLE
Add some es5 features to babel-preset-env

### LINKED ISSUE #6779: @babel/env with useBuiltIns does not include string iterator on IE9
### Input Code

```js
import '@babel/polyfill'
const [foo, bar] = 'fb'
console.log(foo, bar)
```

### Babel/Babylon Configuration (.babelrc, package.json, cli command)

```js
{
  "preset": [
    ["@babel/env", {
      "useBuiltIns": "entry",
      "forceAllTransforms": true,
      "shippedProposals": true,
      "targets": {
        "ie": 9,
        "chrome": 49,
        "edge": 13,
        "firefox": 45,
        "safari": 9
      }
    }]
  ]
}
```

### Expected Behavior
Should print `f` `b`

### Current Behavior
Throws `TypeError` in IE (any version) inside `slicedToArray`

### Your Environment

| software         | version(s)
| ---------------- | -------
| Babel            |  7.0.0-beta.31

### PR DESCRIPTION
**Pull request by existentialism**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #6779
| Patch: Bug Fix?          | n
| Major: Breaking Change?  | n
| Minor: New Feature?      | y
| Tests Added + Pass?      | y
| Documentation PR         | 
| Any Dependency Changes?  |

### CODE REVIEW

File: experimental/babel-preset-env/data/built-in-features.js [hunk 1]
```diff
@@ -1,3 +1,22 @@
+const es5 = {
+  "es6.array.every": "Array methods / Array.prototype.every",
```
Comment (nicolo-ribaudo) [inline experimental/babel-preset-env/data/built-in-features.js]: Shouldn't these be `es5.array.*`? (I have never looked at `preset-env`'s source code, so I may be wrong)

Comment (existentialism) [reply experimental/babel-preset-env/data/built-in-features.js]: These are the names of the matching [core-js modules](https://github.com/zloirock/core-js/blob/master/es5/index.js)

**Comment by nicolo-ribaudo** [comment]
Should es5 polyfills be enabled by default? Or should `env` with no target still mean es2015+?

**Comment by existentialism** [comment]
@nicolo-ribaudo good question.

Right now, with `useBuiltIns: entry`, no target for polyfills includes the whole core-js kitchen sink, so probably ok to maintain the same behavior?

Which leaves just the behavior in `usage` without targets. My initial thought is that it's fine too? Thoughts?

(Also reminds me to update some `useBuiltIns: usage` mappings for these as well.)

**Comment by danez** [comment]
lgtm, but why do we add them?

**Comment by existentialism** [comment]
Updated after landing https://github.com/babel/babel/pull/6755.

@danez it's come up a few times in Slack and various issues, and I feel like it brings completeness/parity with using the entire `babel-polyfill`, as things like `array.sort` are a good example of a polyfill mapping that is needed up until Chrome 63 (which is relatively recent).

File: packages/babel-preset-env/data/built-ins.json [hunk 2]
```diff
@@ -1,4 +1,204 @@
...
+    "android": "4",
+    "ios": "6",
+    "electron": "1.1"
+  },
+  "es6.array.sort": {
+    "chrome": "63",
+    "opera": "50",
+    "edge": "12",
+    "firefox": "5",
+    "safari": "10.1",
```
Comment (Kovensky) [inline packages/babel-preset-env/data/built-ins.json]: This is a bit annoying... according to compat-table, Safari actually _regressed_ on this, and Safari 11 has the same problem as Chrome 62 :cry:

File: packages/babel-preset-env/data/built-ins.json [hunk 3]
```diff
@@ -1,4 +1,204 @@
...
+    "opera": "12",
+    "edge": "12",
+    "firefox": "4",
+    "safari": "5.1",
+    "node": "0.1",
+    "ie": "9",
+    "android": "4",
+    "ios": "7"
+  },
+  "es6.string.iterator": {
```
Comment (Kovensky) [inline packages/babel-preset-env/data/built-ins.json]: I'm pretty sure this doesn't count as "ES5 support" :)

But it's good that this is being fixed; I'm explicitly importing `core-js/fn/string/iterator` to work around it.

File: packages/babel-preset-env/data/built-ins.json (hunk 2 above)
Comment (yavorsky) [reply packages/babel-preset-env/data/built-ins.json]: Maybe it's worth to support ranges there?

**Comment by Rebolon** [comment]
@existentialism How to use it if i want to add Array.for-each shim into my transpiled code ?
I tried on this project https://github.com/Rebolon/tuto-babel-transpile
But i failed, so i used webpack and core-js to help me, but i'm not sure it's the best solution.

Thanks

**Comment by existentialism** [comment]
@Rebolon these changes landed `master`, so you'll need to upgrade all your babel-related dependencies to the 7.0-betas (@babel/preset-env, etc)

**Comment by Rebolon** [comment]
yep i tried with @babel/* on the beta 42 but i didn't find the right way to
do

Benjamin RICHARD
E-mail : benjamin-richard@orange.fr <benjamin-richard@orange.fr>
about.me/benjamin.richard/

2018-03-20 14:55 GMT+01:00 Brian Ng <notifications@github.com>:

**Comment by bekicot** [comment]
@Rebolon Have you fix it already? Encountering the sam issue :(

**Pull request merged**